### DISCUSSION TITLE
Lesson 7 Part 12 Error (7576): Undeclared identifier vm (HelperConfig.s.sol)

### DISCUSSION THREAD

**Discussion by DuncanDuMond** [Cyfrin/foundry-full-course-cu#384] [Q&A] [answered]
In this section, I executed forge test --fork-url $SEPOLIA_RPC_URL on HelperConfig.s.sol and get this error:

[⠰] Compiling...
[⠒] Compiling 3 files with 0.8.19
[⠑] Solc 0.8.19 finished in 166.35ms
Error: 
Compiler run failed:
Error (7576): Undeclared identifier.
  --> script/HelperConfig.s.sol:58:9:
   |
58 |         vm.startBroadcast();
   |         ^^

Error (7576): Undeclared identifier.
  --> script/HelperConfig.s.sol:63:9:
   |
63 |         vm.stopBroadcast();
   |         ^^

I am puzzled why this error is coming up when I have no issues on the DeployFundMe.s.sol file. Here is my code:

// SPDX-License-Identifier: MIT

// 1. Deploy mocks when we are on a local anvil chain
// 2. Keep track of contract address across different chains
// Sepolia ETH / USD Address
// Mainnet ETH / USD Address

pragma solidity ^0.8.19;

import {MockV3Aggregator} from "../test/mocks/MockV3Aggregator.sol";
import {Script} from "forge-std/Script.sol";

contract HelperConfig {
    // If we are on a local anvil, we deploy mocks.
    // Otherwise, grab the existing address from the live network
    NetworkConfig public activeNetworkConfig;

    uint8 public constant DECIMALS = 8;
    int256 public constant INITIAL_PRICE = 2000e8;

    struct NetworkConfig {
        address priceFeed; // ETH/USD price feed address
    }

    constructor() {
        if (block.chainid == 11155111) {
            activeNetworkConfig = getSepoliaEthConfig();
        } else if (block.chainid == 1) {
            activeNetworkConfig = getMainnetEthConfig();
        } else {
            activeNetworkConfig = getOrCreateAnvilEthConfig();
        }
    }

    function getSepoliaEthConfig() public pure returns (NetworkConfig memory) {
        // price feed address
        NetworkConfig memory sepoliaEthConfig = NetworkConfig({
            priceFeed: 0x694AA1769357215DE4FAC081bf1f309aDC325306
        });
        return sepoliaEthConfig;
    }

    function getMainnetEthConfig() public pure returns (NetworkConfig memory) {
        // price feed address
        NetworkConfig memory ethConfig = NetworkConfig({
            priceFeed: 0x5f4eC3Df9cbd43714FE2740f5E3616155c5b8419
        });
        return ethConfig;
    }

    function getOrCreateAnvilEthConfig() public returns (NetworkConfig memory) {
        // Check to see if we set an active network config
        if (activeNetworkConfig.priceFeed != address(0)) {
            return activeNetworkConfig;
        }
        // 1. Deploy the mocks
        // 2. Return the mock address
        vm.startBroadcast();
        MockV3Aggregator mockPriceFeed = new MockV3Aggregator(
            DECIMALS,
            INITIAL_PRICE
        );
        vm.stopBroadcast();

        NetworkConfig memory anvilConfig = NetworkConfig({
            priceFeed: address(mockPriceFeed)
        });
        return anvilConfig;
    }
}

Please advise on a solution recuse I have no answers on how to fix this.

**Comment by ChefAharoni** [comment, marked as answer]
You need to inherit all the functionality from `Script.sol`, like so:
```solidity 
contract HelperConfig is Script {
...
```

That way, you can access all the functions that are inside Script.

**Comment by DuncanDuMond** [comment]
It works now. Thanks!

**Comment by 0xKumarShivam** [comment]
it worked thanks.
